CODING_RULES.md

```diff
@@ -216,3 +216,96 @@ Not part of the caption anymore etc.
+
+- implement item1
+- improve item2
+- remove item3
+- fix item4
+- item5 not yet working
+```
+
+### commit, push, pull
+
+These guidelines mostly apply when working in a feature branch.
+
+Commit, push:
+- work freely in your local branch (including using micro-commits, etc.).
+- shrink contribution's history to a few significant commits *before pushing* (avoid pushing micro-commits or successive versions of same code to the remote)
+  - can use `git commit --amend`
+  - can squash local commits with `git rebase -i local_branch_base_commit` to shrink the yet-unpushed commits from `local_branch_base_commit`. Typically keep the first commit as `pick`, don't change order of commits, turn the next ones to `squash` to merge them in a single commit.
+  - stick to squashing the latest commits (unless you really know what you are doing)
+
+Pull:
+- can always use `git pull --rebase` (or add it to configuration to apply by default). No danger, it only rebases local yet-unpushed commits on top of new pulled remote commits.
+
+### update, merge
+
+These guidelines mostly apply when updating a feature branch or merging it to `develop`.
+
+Update a feature branch with latest version of `develop`:
+- nice to rebase feature branch from develop (`git rebase develop`) as explained below, rather than merging develop in feature branch (`git merge develop`)
+- caveat: don't mix rebase and merging for successive updates of a feature branch !
+
+Merge:
+- before merging, if too many commits were pushed to the feature branch, squash *remote branch* with a rebase (see below)
+
+### rebase remote branch
+
+These guidelines mostly apply when updating a feature branch or merging it to `develop`.
+- don't rebase on `develop` and `master` branches. Merge to these branches using a merge commit (default in github).
+
+**Warning !**
+- **be careful when using `git rebase` on commits *already pushed to the remote*. Bad operation brings risk of losing or duplicating commit**. Use `git reflog` if lost some commit.
+- use rebase when simple enough, if complicated or doubtful do a merge
+- typically stick to squashing the latest commits (unless you really know what you are doing)
+
+How to rebase a *remote* feature branch (from `develop` or to squash commits):
+- coordinate with other team members
+  * OK to rebase when working alone in feature branch
+  * when other developers work/commit/push in feature branch, ask if OK to rebase now (eg: Discord), and inform after rebasing is done
+- update local clone of `develop` (if rebasing from `develop`) and local clone of `feature_branch`
+- `git checkout feature_branch`
+- if rebasing from `develop` typically use `git rebase -i develop` with default values
+- if squashing feature branch typically use `git rebase -i feature_branch_base_commit`, keep the first commit as `pick`, don't change order of commits, turn the commits you want to squash as `squash`.
+- manually solve conflicts if any to complete rebasing
+- pull before pushing
```

Comment:
Yes you're right, this should be redundant unless something went wrong ... this might be confusing => I would propose to remove it.

Sorry, I had wrongly fused the bits from our discussions.

Reply:
After discussion: removed.